tests/scripts/all.sh

```diff
@@ -76,41 +76,14 @@
 # Notes for maintainers
```

Comment:
We should split both the components and the engine into `all.sh`. So `all.sh` should be something like
```
#!/bin/bash
script_dir=${0##*/}
component_dir=$script_dir
. "$script_dir/test-runner-engine.sh"
for x in "$component_dir"/test-components-*.sh; do
  . "$x"
done
test_runner_main "$@"
```
with `test_runner_main` containing all the top-level code.